doc/sphinx/practical-tools/coq-commands.rst

```diff
@@ -148,39 +169,53 @@ except that ``space_overhead`` is set to 120 and ``minor_heap_size`` is set to 3
+Command line options
+~~~~~~~~~~~~~~~~~~~~
 
 The following command-line options are recognized by the commands ``coqc``
 and ``coqtop``, unless stated otherwise:
 
 :-I *directory*, -include *directory*: Add physical path *directory*
-  to the OCaml loadpath.
+  to the OCaml loadpath, which is needed to load OCaml object code files
+  (``.cmo`` or ``.cmxs``).  Subdirectories are not included.
+  See the command :cmd:`Declare ML Module`.
+
+.. TODO PR: is that right about Declare ML Module? it's not a directory like -I
 
   .. seealso::
 
-     :ref:`names-of-libraries` and the
-     command Declare ML Module Section :ref:`compiled-files`.
-:-Q *directory dirpath*: Add physical path *directory* to the list of
-  directories where Coq looks for a file and bind it to the logical
-  directory *dirpath*. The subdirectory structure of *directory* is
-  recursively available from Coq using absolute names (extending the
-  :n:`@dirpath` prefix) (see Section :ref:`qualified-names`). Note that only those
-  subdirectories and files which obey the lexical conventions of what is
-  an :n:`@ident` are taken into account. Conversely, the
-  underlying file systems or operating systems may be more restrictive
-  than Coq. While Linux’s ext4 file system supports any Coq recursive
-  layout (within the limit of 255 bytes per filename), the default on
-  NTFS (Windows) or HFS+ (MacOS X) file systems is on the contrary to
-  disallow two files differing only in the case in the same directory.
-
-  .. seealso:: Section :ref:`names-of-libraries`, :ref:`libraries-and-filesystem` and :cmd:`Require`.
-:-R *directory dirpath*: Similar to ``-Q`` *directory dirpath*, but allow using
-  :cmd:`Require` with a partially qualified name.
-
-  .. seealso:: Section :ref:`names-of-libraries`, :ref:`libraries-and-filesystem` and :cmd:`Require`.
+     The :cmd:`Declare ML Module` command.
+
+.. _-Q-option:
+
+:-Q *directory dirpath*: Makes the `.vo` files in a :term:`package` available for
+  loading with the :cmd:`Require` command by adding new entries to the :term:`load path`.
+  The entries map the
+  :term:`logical path` *dirpath* to the physical path *directory*.  Then it
+  recursively adds loadpath entries for subdirectories.  For example, `-Q . Lib`
+  may add the logical path `Lib.SubDir.File`, which maps to the file
+  `./SubDir/File.vo`.
+
+  Only subdirectories and files that follow the lexical conventions for
+  :n:`@ident`\s are included.  Subdirectories named ``CVS`` or
+  ``_darcs`` are excluded. Some operating systems or file systems are
+  more restrictive.  For example, Linux’s ext4 file system limits filenames
+  to 255 bytes.  The
+  default on NTFS (Windows) and HFS+ (MacOS X) file systems is to
+  disallow two files in the same directory with names that differ only in their
+  case.
+
+  Loading files from packages made available with `-Q` must include
+  the :term:`logical name` of the package in `From` clause of the :cmd:`Require`
```

Comment:
or the Require may use the fully qualified name, eg `Require Ltac2.Ltac2`

Reply:
Also, I didn't mention possibilities such as `-R . Foo.Bar`, thinking they are not very useful.  (My impression is that we've made some of the config details a bit too general, giving lots of corner cases that are not that useful.  I've tried not to dwell on these, they mostly add complexity/confusion.)